src/Riok.Mapperly/Descriptors/MappingBuilders/EnumerableMappingBuilder.cs

```diff
@@ -63,6 +63,11 @@ public static class EnumerableMappingBuilder
         if (immutableLinqMapping is not null)
             return immutableLinqMapping;
 
+        // if target is a type that takes IEnumerable in its constructor
+        if (ctx.Target.ImplementsGeneric(ctx.Types.StackT, out _)
+            || ctx.Target.ImplementsGeneric(ctx.Types.QueueT, out _))
```

Comment:
On a second thought: couldn't we generalize this to check if such a ctor is available instead of checking only those types?

Reply:
It could cause some unexpected behaviour. It can be slower to use constructors in some cases. For example `List<T>` does not check for `IListProvider` so a foreach loop with `EnsureCapacity` might be faster (`List<T>` can't hit this section of code).